wallet/src/state.rs

```diff
@@ -227,6 +227,15 @@ impl ClientState {
         }
     }
 
+    fn chain_id(&self) -> Result<String, anyhow::Error> {
+        let chain_params = self.chain_params();
+        if chain_params.is_none() {
+            return Err(anyhow::anyhow!("chain_params not set on state"));
+        }
+
+        Ok(chain_params.unwrap().chain_id.clone())
```

Comment:
tioli: this could be simplified to `Ok(self.chain_params().ok_or_else(|| anyhow!(...))?.chain_id.clone())`, but it doesn't really matter here.